restapi/client.go

```diff
@@ -418,6 +418,16 @@ func newS3BucketClient(claims *models.Principal, bucketName string, prefix strin
 	return s3Client, nil
 }
 
+// pathJoinFinalSlash - like path.Join() but retains trailing slashSeparator of the last element
+func pathJoinFinalSlash(elem ...string) string {
+	if len(elem) > 0 {
+		if strings.HasSuffix(elem[len(elem)-1], "/") {
```

Comment:
This is not necessary, HasSuffix returns true of false for the whole string, you don't need to get the lat character for this, https://go.dev/play/p/BlT26DEoXXN

 ```suggestion
		if strings.HasSuffix(elem, "/") {
```

Reply:
It is an optimization @bexsoft to avoid through the entire string - look for only what you are interested